appcenter-data/android/src/main/java/com/microsoft/appcenter/reactnative/data/AppCenterReactNativeDataModule.java

```diff
@@ -56,8 +60,12 @@ public String getName() {
     }
 
     @ReactMethod
-    public void read(String documentId, String partition, int ttl, final Promise promise) {
-        Data.read(documentId, JsonElement.class, partition, new ReadOptions(ttl)).thenAccept(new AppCenterConsumer<DocumentWrapper<JsonElement>>() {
+    public void read(String documentId, String partition, ReadableMap readOptionsMap, final Promise promise) {
+        ReadOptions readOptions = new ReadOptions(TimeToLive.DEFAULT);
```

Comment:
avoid allocate an object if being replaced, move it to else.

Reply:
Fixed